src/Tracker.js

```diff
@@ -0,0 +1,115 @@
+import { useEffect, useState } from 'react'
+import axios from 'axios'
+import L from 'leaflet'
+import {
+  CircleMarker, GeoJSON, MapContainer, Marker, TileLayer
+} from 'react-leaflet'
+import 'leaflet-rotatedmarker'
+import uniqid from 'uniqid'
+
+import {
+  getActivity, getBearing, getLocation, getLineName
+} from './helpers/busTimeApiHelpers'
+import settings from './settings/busSettings'
+import { busTimeAPI, proxyURL } from './settings/busTimeSettings'
+import {
+  mapAttribution, mapboxURL, mapCenter, mapZoom
+} from './settings/mapSettings'
+
+import 'leaflet/dist/leaflet.css'
+import './Tracker.css'
+
+function Tracker() {
+  const [busData, setBusData] = useState({})
+
+  const getBusData = () => {
+    const apiRequests = []
+    const busLines = []
+
+    Object.values(settings).map((bus) => (
+      apiRequests.push(
+        axios.get(proxyURL + encodeURIComponent(busTimeAPI + bus.line))
+          .then((response) => busLines.push(response))
+      )
+    ))
+
+    Promise.all(apiRequests).then(() => {
+      busLines.forEach((line) => {
+        const activity = getActivity(line)
+        const lineName = getLineName(activity)
+
+        const buses = activity.map((bus) => {
+          const location = getLocation(bus)
+          const bearing = getBearing(bus)
+
+          return [[location.Latitude, location.Longitude], bearing]
+        })
+
+        setBusData((prevBusData) => ({
+          ...prevBusData,
+          [lineName]: { name: lineName, buses }
+        }))
+      })
+    })
+      .catch((error) => { throw new Error(error) })
+  }
+
+  const drawBus = (line, position) => {
+    const [location, bearing] = position
+
+    return (
+      <Marker
+        icon={L.icon({
+          iconSize: [40, 40],
+          iconUrl: settings[line].marker
+        })}
+        key={uniqid()}
+        position={location}
+        rotationAngle={-bearing}
+        rotationOrigin="center center"
+      />
+    )
+  }
+
+  const drawRoute = (line) => (
+    <GeoJSON
+      data={line.route}
+      key={uniqid()}
+      style={{ color: `#${line.color}33` }}
```

Comment:
{{ thing }}

Reply:
Ah right. JS + React style object. It looked like your internal templating language didn't render properly. 😆